### PR TITLE
feat(ban): add endpoints for ban handling

### PR DESCRIPTION
**Pull request by Antreesy**
### ☑️ Resolves

* Ref #12293
* Provide a minimal frontend to test feature

> [!TIP]
> To test handling a page navigation:
> - create user with id `banned`
> - create conversation with name `ban user`
> - try to join as user

### 🖼️ Screenshots / Screencasts

Ban | -
-- | --
image | image
List | -
image | -
image | image
Page | -
image | -

### 🏁 Checklist

- [x] 🌏 Tested with Chrome, Firefox and Safari or should not be risky to browser differences
- [x] 🖥️ Tested with Desktop client or should not be risky for it 
- [x] 🖌️ Design was reviewed, approved or inspired by the design team
- [x] ⛑️ Tests are included or not possible
- [ ] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is not required

### CODE REVIEW

File: src/components/ConversationSettings/BanSettings/BannedItem.vue [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+				t('spreed', 'Note: {note}', { note: this.ban.internalNote }),
+			]
+		},
+	},
+}
+</script>
+
+<style lang="scss" scoped>
+
+.ban-item {
```
Comment (Antreesy) [inline src/components/ConversationSettings/BanSettings/BannedItem.vue:70]: Maybe add some spacing and dividers, to distinguish items?
Plus bold caption
image

**Comment by Antreesy** [comment]
We also forgot to add error handling for banned users/guests (but we can merge this first and do a follow-up for the other part):
- create user with id `banned`
- create conversation with name `ban user`
- try to join as user
- you should see 403 Error with `response.data.ocs: {"data": {  "error": "ban" }}`
- probably we need to redirect to another page (like 404 - Not Found, but specific for 403 / bans)

**Comment by nickvergessen** [comment]
> Should be two requests to the server, or could be done with a single entrypoint?

I think I would add the options to the existing endpoint.

**Review by marcoambrosini — APPROVED** [review]
Sorry for late reply, looks good to me :) 
Maybe we could repeat the name instead of referring to "participant" in the checkbox text

File: src/store/participantsStore.js [hunk 2]
```diff
@@ -560,12 +561,25 @@ const actions = {
...
+					actorId: attendee.actorId,
+					actorType: attendee.actorType,
+					internalNote,
+				})
+				showSuccess(t('spreed', 'Participant is banned successfully'))
+			} catch (error) {
+				showError(t('spreed', 'Error while banning the participant'))
+				console.error('Error while banning the participant: ', error)
+			}
+		}
```
[the remark this answers is not in the document]
Comment (Antreesy) [reply src/store/participantsStore.js:582]: As soon as API will be adjusted, we have to update that code block anyway

**Review by DorraJaouad — APPROVED** [review]
Tested 🦅

**Pull request merged**